app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :author, foreign_key: 'author_id', class_name: 'User'
+  belongs_to :post, foreign_key: 'post_id', class_name: 'Post'
+
+  def update_comment_counts
+    post = Post.find(post_id)
+    post.comment_count = Comment.where(post_id: post.id).count
+    post.save
+  end
```

Comment:
[OPTIONAL]
- Just like before in `user.rb`, you can add `after_save :update_comment_counts`